### PR TITLE
[WIP]Add skip to content- keyboard accessibility

### LINKED ISSUE #314: [UI] Keyboard accessibility
STR:

1. Navigate to https://taskcluster-web.netlify.com/tasks/index
2. Click the tab key until the focus goes to the first namespace item:

screen shot 2019-02-26 at 3 13 56 pm

**Problem**: The is no way to jump the app bar with the tab key.

**Expected**: A "Skip to Content" button appears after the first usage of the tab key allowing the user to skip navigating the app bar if they press on enter.

**Actual**: The user is required to press the tab key 6 times before having the focus on a list item

You can see it live in action on https://github.com/taskcluster/taskcluster. Click tab and you will see a hidden button pop up.

### PR DESCRIPTION
**Pull request by Aditya-Kolla**
This is a WIP PR.
Temporary position and look.
Created a new component for SkipNavigation.
fixes #314

### CODE REVIEW

**Comment by Aditya-Kolla** [comment]
@helfi92 Can you help me understand the styling a bit. 
I have figured it out on plain css but not able to translate it to our project.

File: ui/src/components/SkipNavigation/index.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React, { Component } from 'react';
+import { Typography } from '@material-ui/core';
+import Link from '../../utils/Link';
+
+export default class SkipNavigation extends Component {
+  render() {
+    const { href, text } = this.props;
+
+    return (
+      <Link to={href}>
```
Comment (helfi92) [inline ui/src/components/SkipNavigation/index.jsx]: I think you can use a Button from `src/components/Button` instead and have an `onClick` that jumps to the first focusable element inside `<main>`.

Comment (Aditya-Kolla) [reply ui/src/components/SkipNavigation/index.jsx]: Alright. Will do.

**Comment by Aditya-Kolla** [comment]
@helfi92 I was thinking of pushing it out of the screen space but the clipping method which you showed seems like a better idea. I'll implement that :)

**Comment by helfi92** [comment]
> I was thinking of pushing it out of the screen space but the clipping method which you showed seems like a better idea. I'll implement that :)

The clipping idea is not mine. I have never used it. We can try using it if it makes sense :)

**Comment by Aditya-Kolla** [comment]
> The clipping idea is not mine. I have never used it. We can try using it if it makes sense :)

Ah, fair enough.

File: ui/src/components/SkipNavigation/index.jsx (hunk 1 above)
Comment (Aditya-Kolla) [reply ui/src/components/SkipNavigation/index.jsx]: I have done these changes.

**Comment by Aditya-Kolla** [comment]
@helfi92 I've changed the Link to Button and added css to make it invisible by default and appear when it gets focus. 
I realised that to use clip I need to define a canvas so implemented my initial approach.
Yet to add button onClick which I'll be doing as soon as I wake up tomorrow morning :sleeping:

**Comment by helfi92** [comment]
Awesome, good night :)

**Comment by Aditya-Kolla** [comment]
Hey @helfi92, I started work on the onClick and was wondering how could I get the first focus-sable element in the main tag? 
Will I need to use refs?

**Comment by helfi92** [comment]
> I started work on the onClick and was wondering how could I get the first focus-sable element in the main tag

I think it should be fine to simply jump to the `<main>` element. You can use the DOM API to have access. Maybe something like `document.querySelector('main')`.

**Comment by Aditya-Kolla** [comment]
Hello, sorry for not being able to work on this lately. 
In this commit, I changed the skip button color to red so it brings a little contrast.
And the focus changes to main tag on button click.
Please review.

**Comment by djmitche** [comment]
Hassan will be back next week and can look at this.

**Comment by helfi92** [comment]
@Aditya-Kolla How is it going? Let me know if you had any questions for me :)

**Comment by Aditya-Kolla** [comment]
> @Aditya-Kolla How is it going? Let me know if you had any questions for me :)

Hello, sorry I got a bit busy. I'll resume work on this :D

File: ui/src/components/SkipNavigation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { string } from 'prop-types';
+import { withStyles } from '@material-ui/core';
+import Button from '../Button';
+
+@withStyles({
+  skipButton: {
+    position: 'absolute',
+    left: '-999px',
```
Comment (helfi92) [inline ui/src/components/SkipNavigation/index.jsx]: `left: -999,`

Comment (Aditya-Kolla) [reply ui/src/components/SkipNavigation/index.jsx]: Done. Mind if I ask what the difference between the two is? Thank you :D

Comment (helfi92) [reply ui/src/components/SkipNavigation/index.jsx]: There is no difference except fewer characters and consistency througout the codebase. :)

**Comment by helfi92** [comment]
@Aditya-Kolla I accidentally pushed a commit. Do you mind creating a new PR with your changes, please. Sorry for that.

**Pull request closed without merging**